Hummingbird.docc/Articles/Recommendations.md

```diff
@@ -0,0 +1,342 @@
+## Dependencies
+
+Hummingbird won't enforce any dependencies, or provide special treatment for dependencies on a framework level. However, we'll provide some guidelines and tips for _common_ dependencies and how to use them.
+
+## Project Structure
+
+The project structure is a recommended project layout, defining what modules are recommended to create and how these modules are structured in terms of folders and files.
+
+### Module Structure
+
+Every Hummingbird projects neds a primary executable target. This is the **MyApp** target in the example below.
+
+If a user has an OpenAPI specification for their API, a separate OpenAPI module is recommended based on [swift-openapi-generator](https://github.com/apple/swift-openapi-generator). This module will be used to generate the API server code.
+
+```
+├── Sources
+│   ├── MyApp
+│   └── MyAppOpenAPI # Optional
+│       
+├── Tests
+│   └── MyAppTests
+│
+└── Package.swift
+```
+
+## App Module Structure
+
+The **MyApp** module is the primary module for the project. It contains the entry point for the application in **App.swift**, and has folders for **routes**, **database**, **repositories**, **services**, **middleware** and **config**.
+
+
+```
+├── Routes
+│   ├── ...Routes.swift
+│   └── BuildRouter.swift
+│
+├── Database
+│   ├── Models
+│   └── Migrations
+│
+├── Repositories
```

Comment:
I don't think we should have a repositories folder. If this is aimed at individuals who are new to web development using Swift I'm not sure they'll know what this involved. If we feel this is the correct place we can add a Repositories section later in the guidelines